KAA-1362: [C++ SDK] Make FindBotan cross platform

vasner opened the pull request:
Fix missed support of OS X in this script;
Deprecate naming libraries like lib*;
Add library check like in "FindAvro.cmake";
Make up code style in Modules cmake scripts;

Reviewers: @rasendubi and @forGGe

vasner committed:
KAA-1362: [C++ SDK] Make FindBotan cross platform
Fix missed support of OS X in this script;
Deprecate naming libraries like lib*;
Add library check like in "FindAvro.cmake";
Make up code style in Modules cmake scripts;

rasendubi commented on client/client-multi/client-cpp/Modules/FindAvro.cmake:
```diff
@@ -28,50 +28,47 @@
+     ${CMAKE_INCLUDE_PATH})
 
-if(Avro_USE_STATIC_LIBS)
+if(AVRO_USE_STATIC_LIBS)
```
Breaks API

vasner replied:
Fixed.

vasner committed:
KAA-1362: [C++ SDK] Make FindBotan cross platform
Fix API static lib mane convention;

rasendubi commented on client/client-multi/client-cpp/Modules/FindBotan.cmake:
```diff
@@ -16,12 +16,15 @@
 # -*- mode: cmake; -*-
 # - Try to find libbotan include dirs and libraries
 # Usage of this module as follows:
+# * find_package(Botan)
```
Remove star. It was used below as a list item mark.

rasendubi commented on client/client-multi/client-cpp/Modules/FindBotan.cmake:
```diff
@@ -30,37 +33,46 @@ if(NOT WIN32)
     ${CMAKE_FIND_ROOT_PATH}/include
-    ${BOTAN_INCLUDE_DIRS})
+    ${BOTAN_INCLUDE_DIRS}
+    ${CMAKE_INCLUDE_PATH})
```
`CMAKE_INCLUDE_PATH` is implicit

vasner replied:
What do you want to do with it? Sorry, but there I just copy-paste FindAvro.cmake

rasendubi replied:
That means you may remove it here as it doesn't change anything.

vasner replied:
Fixed.

rasendubi commented on client/client-multi/client-cpp/Modules/FindBotan.cmake:
```diff
@@ -30,37 +33,46 @@ if(NOT WIN32)
 else(WIN32)
     if(Botan_USE_STATIC_LIBS)
-        set(BOTAN_LIBRARY_NAMES ${BOTAN_LIBRARY_NAMES} libbotan.a libbotan-1.11.a)
+        set(BOTAN_LIBRARY_NAMES ${BOTAN_LIBRARY_NAMES} botan botan-1.11)
```
Please, leave `lib` and `.a` for static libs. Otherwise, it's not different from dynamic ones

vasner committed:
KAA-1362: [C++ SDK] Make FindBotan cross platform
Remove implicit CMAKE_INCLUDE_PATH;
Make up comments;
Leave lib.a style for Botan static libraries;
